src/main/scala/com/cognite/sdk/scala/v1/fdm/instances/InstancePropertyValue.scala

```diff
@@ -228,6 +223,34 @@ object InstancePropertyValue {
     result.getOrElse(Left(DecodingFailure(s"Missing Instance Property Type", c.history)))
   }
 
+  private def isDirectRelation(v: Json) =
+    v.asObject.exists(obj => obj.contains("externalId") && obj.contains("space") && obj.size === 2)
+  private def decodeDirectRelation(
+      v: Json,
+      c: HCursor
+  ): Either[DecodingFailure, DirectRelationReference] =
+    v.asObject
+      .map(obj =>
+        for {
+          spaceJson <- obj("space")
+            .toRight(DecodingFailure("Missing space property in direct relation object", c.history))
+          space <- spaceJson.asString
+            .toRight(DecodingFailure("Space isn't string in direct relation object", c.history))
+          externalIdJson <- obj("externalId").toRight(
+            DecodingFailure("Missing externalId in direct relation object", c.history)
+          )
+          externalId <- externalIdJson.asString.toRight(
+            DecodingFailure("ExternalId isn't string in direct relation object", c.history)
+          )
+        } yield DirectRelationReference(
+          space,
+          externalId
+        )
+      )
+      .getOrElse(
+        Left(DecodingFailure("Could not deserialize into direct relation object", c.history))
```

Comment:
note: this one shouldn't be ever hit as `v.asObject.map()` is always `Some()`  for `v.isObject / v.asObject.exists()`, but error message still makes sense and maybe not worth the effort to try and combine all isObjects checks into single place :slightly_smiling_face:

Reply:
I knew that but you kind of need to handle the orElse here and I was thinking it's best to leave a helpful error message if we ever change the code or mess it up somehow.